src/garage/torch/modules/cnn_module.py

```diff
@@ -0,0 +1,252 @@
+            hidden_layers = nn.Sequential()
+
+            if isinstance(self._paddings, (list, tuple)):
+                padding = self._paddings[index]
+            elif isinstance(self._paddings, int):
+                padding = self._paddings
+
+            # conv 2D layer
+            conv_layer = _conv(in_channels=prev_channel,
+                               out_channels=channel,
+                               kernel_size=kernel_size,
+                               stride=stride,
+                               padding=padding)
+            self._hidden_w_init(conv_layer.weight)
+            self._hidden_b_init(conv_layer.bias)
+            hidden_layers.add_module('conv_{}'.format(index), conv_layer)
+
+            # layer normalization
+            if self._layer_normalization:
+                hidden_layers.add_module('layer_normalization',
+                                         nn.LayerNorm(channel))
+
+            # non-linear function
+            if self._hidden_nonlinearity:
+                hidden_layers.add_module(
+                    'non_linearity', _NonLinearity(self._hidden_nonlinearity))
+
+            # max-pooling
+            if self._max_pool:
+                hidden_layers.add_module(
+                    'max_pooling',
+                    nn.MaxPool2d(kernel_size=self._pool_shape,
+                                 stride=self._pool_stride))
+
+            self._cnn_layers.append(hidden_layers)
+            prev_channel = channel
+
+
+def _conv(in_channels,
+          out_channels,
+          kernel_size,
+          stride=1,
+          padding=0,
+          padding_mode='zeros',
+          dilation=1,
+          bias=True):
+    """Helper function for performing convolution.
+
+    Args:
+        in_channels (int):
+            Number of channels in the input image
+        out_channels (int):
+            Number of channels produced by the convolution
+        kernel_size (int or tuple):
+            Size of the convolving kernel
+        stride (int or tuple): Stride of the convolution.
+            Default: 1
+        padding (int or tuple): Zero-padding added to both sides
+            of the input. Default: 0
+        padding_mode (string): 'zeros', 'reflect', 'replicate'
+            or 'circular'. Default: 'zeros'
+        dilation (int or tuple): Spacing between kernel elements.
+            Default: 1
+        bias (bool): If True, adds a learnable bias to the output.
+            Default: True
+
+    Returns:
+        torch.Tensor: The output of the 2D convolution.
+
+    """
+    return nn.Conv2d(in_channels=in_channels,
+                     out_channels=out_channels,
+                     kernel_size=kernel_size,
+                     stride=stride,
+                     padding=padding,
+                     padding_mode=padding_mode,
+                     dilation=dilation,
+                     bias=bias)
+
+
+class _NonLinearity(nn.Module):
```

Comment:
This should probably go in `garage.torch._functions`, since it's definitely wanted elsewhere.